desktop/updater/auto-updater/index.js

```diff
@@ -5,58 +5,64 @@
  */
 const EventEmitter = require('events').EventEmitter;
 const util = require('util');
-const electron = require( 'electron' );
-const autoUpdater = electron.autoUpdater;
-const dialog = electron.dialog;
+const { dialog } = require('electron');
+const { autoUpdater } = require('electron-updater');
+const sanitizeHtml = require('sanitize-html');
+
+const config = require('../../config');
 
 /**
  * Internal dependencies
  */
-const AppQuit = require( '../..//app-quit' );
-
-function AutoUpdater( url ) {
-	this.hasPrompted = false;
-	this.url = url;
+const AppQuit = require('../../app-quit');
 
-	autoUpdater.on( 'error', () => this.emit( 'end' ) );
-	autoUpdater.on( 'update-not-available', () => this.emit( 'end' ) );
-	autoUpdater.on( 'update-downloaded', this.onDownloaded.bind( this ) );
+function AutoUpdater() {
+  this.hasPrompted = false;
 
-	try {
-		autoUpdater.setFeedURL( url );
-	} catch (e) {
-		console.log( e.message );
-	}
+  autoUpdater.on('error', () => this.emit('end'));
+  autoUpdater.on('update-not-available', () => this.emit('end'));
+  autoUpdater.on('update-downloaded', this.onDownloaded.bind(this));
 }
 
 util.inherits(AutoUpdater, EventEmitter);
 
 AutoUpdater.prototype.ping = function() {
-	autoUpdater.checkForUpdates();
+  autoUpdater.checkForUpdates();
+};
+
+AutoUpdater.prototype.onDownloaded = function(info) {
+  const releaseNotes = this.formatGithubReleaseNotes(info.releaseNotes);
+
+  const updateDialogOptions = {
+    buttons: ['Update & Restart', 'Cancel'],
+    title: 'Update Available',
+    message: `${config.name} ${info.releaseName}`,
+    detail: releaseNotes,
+  };
+
+  if (this.hasPrompted === false) {
+    this.hasPrompted = true;
+
+    dialog.showMessageBox(updateDialogOptions, button => {
+      this.hasPrompted = false;
+
+      if (button === 0) {
+        AppQuit.allowQuit();
+        autoUpdater.quitAndInstall();
+      } else {
+        this.emit('end');
+      }
+    });
+  }
 };
 
-AutoUpdater.prototype.onDownloaded = function( event, releaseNotes, releaseName, releaseDate, updateUrl, quitAndUpdate ) {
-	const updateDialogOptions = {
-		buttons: [ 'Update & Restart', 'Cancel' ],
-		title: 'Update Available',
-		message: releaseName,
-		detail: releaseNotes
-	};
-
-	if ( this.hasPrompted === false ) {
-		this.hasPrompted = true;
-
-		dialog.showMessageBox( updateDialogOptions, button => {
-			this.hasPrompted = false;
-
-			if ( button === 0 ) {
-				AppQuit.allowQuit();
-				quitAndUpdate();
-			} else {
-				this.emit( 'end' );
-			}
-		} );
-	}
+AutoUpdater.prototype.formatGithubReleaseNotes = function(releaseNotes) {
+  return sanitizeHtml(releaseNotes, {
+    allowedTags: ['br', 'li'],
+    allowedAttributes: [],
+  })
+    .replace(/<br \/>/g, '\n')
+    .replace(/<li>(.*)<\/li>/g, '- $1');
```

Comment:
```html
<ul>
  <li> Major Changes
    <ul>
      <li>New AutoUpdater</li>
      <li>RegExps parsing HTML</li>
    </ul>
  </li>
  <li>Minor Changes
    <ul>
      <li>Updated code</li>
    </ul>
  </li>
</ul>
```

```html
<ul>
  <li> Major Changes
    <ul>
       - New AutoUpdater
       - RegExps parsing HTML
    </ul>
  </li>
  <li>Minor Changes
    <ul>
       - Updated code
    </ul>
  </li>
</ul>
```

If we want to perform this replacement we should parse the HTML and replace those nodes otherwise we'll get unexpected results.

Reply:
In the beginning, I thought so too as I wanted to be able to display more complex text structures by e.g. converting the HTML to a cleaned up form of markdown (removing urls, text styles, ...). 

By looking at the way release notes are currently done, I'm just not sure if we wouldn't over engineer this by starting to parse the HTML – instead I went for the IMO simplest solution – but I'm definitely not holding on to this if I should do it differently!

Semi off topic: On the long run, I'd vote for a solution like using `conventional-changelog` and webview enhanced dialogs (once available in electron) to display changelogs.

Reply:
Wouldn't necessarily call a two-level list as a more complex structure, or even this…

```html
<li><p>This spans<br />
two lines</p></li>
```

Also, HTML parsing isn't that hard - the browser does it for us - do we have the browser available here?

In other words, we can use a `TreeWalker` to iterate over the `li` nodes and replace them _and_ their containing `ul` or `ol` nodes with the proper list symbol.

Alternatively we could strip all tags or show the webview as you mention. If we go half-way I'm guessing we'll get crazy unexpected displays.